gpiozero/boards.py

```diff
@@ -1185,3 +1185,86 @@ def on(self):
+        'D': (False, True, True, True, True, False, True),
+        'E': (True, False, False, True, True, True, True),
+        'F': (True, False, False, False, True, True, True),
+        'G': (True, False, True, True, True, True, False),
+        'H': (False, True, True, False, True, True, True),
+        'I': (False, False, False, False, True, True, False),
+        'J': (False, True, True, True, True, False, False),
+        'K': (True, False, True, False, True, True, True),
+        'L': (False, False, False, True, True, True, False),
+        'M': (True, False, True, False, True, False, False),
+        'N': (True, True, True, False, True, True, False),
+        'O': (True, True, True, True, True, True, False),
+        'P': (True, True, False, False, True, True, True),
+        'Q': (True, True, False, True, False, True, True),
+        'R': (True, True, False, False, True, True, False),
+        'S': (True, False, True, True, False, True, True),
+        'T': (False, False, False, True, True, True, True),
+        'U': (False, True, True, True, True, True, False),
+        'V': (False, True, True, True, True, True, False),
+        'W': (False, True, False, True, False, True, False),
+        'X': (False, True, True, False, True, True, True),
+        'Y': (False, True, True, True, False, True, True),
+        'Z': (True, True, False, True, True, False, True),
+        '-': (False, False, False, False, False, False, True),
+        ' ': (False, False, False, False, False, False, False)
+        }
+    
+    def __init__(self, *pins, **kwargs):
+        # 7 segment displays must have 7 or 8 pins
+        if len(pins) < 7 or len(pins) > 8:
+            raise ValueError('7 segment display must have 7 or 8 pins')
+        # Don't allow 7 segments to contain collections
+        for pin in pins:
+            assert not isinstance(pin, LEDCollection)
+        pwm = kwargs.pop('pwm', False)
+        active_high = kwargs.pop('active_high', True)
+        if kwargs:
+            raise TypeError('unexpected keyword argument: %s' % kwargs.popitem()[0])
+        super(SevenSegmentDisplay, self).__init__(*pins, pwm=pwm, active_high=active_high)
+    
+    def display(self, char):
+        char = str(char).upper()
+        if len(char) > 1:
+            raise ValueError('Only a single character can be displayed')
+        if char in SevenSegmentDisplay.layouts:
+            layout = SevenSegmentDisplay.layouts[char]
+            for led in range(7):
+                self[led].value = layout[led]
+        else:
+            raise ValueError('There is no layout for character %s')
+            
+    def display_hex(self, hexnumber):
+        self.display(hex(hexnumber)[2:])
+
+    @property
+    def decimal_point(self):
+        # does the 7seg display have a decimal point (i.e pin 8)
+        if len(self) > 7:
+            return self[7].value 
+        else:
+            raise OutputDeviceError('There is no 8th pin for the decimal point')
```

Comment:
I can't decide whether trying to get the decimal point on a 7-pin SevenSegment should just always return `False`, rather than raising an exception?

(OTOH, trying to set the dp on a 7-pin SevenSegment probably _should_ still raise an exception?)